docs/upgrade.md

```diff
@@ -88,6 +88,18 @@ process, for example:
     dpkg -i matrix-synapse-py3_1.3.0+stretch1_amd64.deb
     ```
 
+# Upgrading to v1.74.0
+
+## Unicode support in user search
+
+This version introduces optional support for an [improved user search dealing with Unicode characters](https://github.com/matrix-org/synapse/pull/14464).
+
+To do so it uses the ICU library through PyICU.
+If you want to take advantage of this feature you need to install PyICU
```

Comment:
note that this is only for people installing from source or PyPI — Docker and Debs not affected.

(can we find out if we include the dependency in docker and debs and say so here?)

Reply:
Hummm we do but I think it's not working: we install the dev package in the build image, but we don't install the runtime package in the runtime image. I'll fix that in an independent PR.

Reply:
Actually it does because it's a dependency of one of our dependency. I'll add an explicit install in case of.

Reply:
For deb `shlibs:Depends` is taking care of that I believe.

Reply:
I've sneak the Dockerfile change here since it is not a real fix but more a `let's be explicit`.